cosmos/x/evm/keeper/processor.go

```diff
@@ -22,45 +22,51 @@ package keeper
 
 import (
 	"context"
+	"fmt"
 
 	sdk "github.com/cosmos/cosmos-sdk/types"
 
-	"pkg.berachain.dev/polaris/eth/core"
 	coretypes "pkg.berachain.dev/polaris/eth/core/types"
 )
 
 // ProcessTransaction is called during the DeliverTx processing of the ABCI lifecycle.
-func (k *Keeper) ProcessTransaction(ctx context.Context, tx *coretypes.Transaction) (*core.ExecutionResult, error) {
+func (k *Keeper) ProcessTransaction(ctx context.Context, tx *coretypes.Transaction) (*coretypes.Receipt, error) {
 	sCtx := sdk.UnwrapSDKContext(ctx)
+	gasMeter := sCtx.GasMeter()
 	// We zero-out the gas meter prior to evm execution in order to ensure that the receipt output
 	// from the EVM is correct. In the future, we will revisit this to allow gas metering for more
 	// complex operations prior to entering the EVM.
-	sCtx.GasMeter().RefundGas(sCtx.GasMeter().GasConsumed(),
+	gasMeter.RefundGas(gasMeter.GasConsumed(),
 		"reset gas meter prior to ethereum state transition")
 
 	// Process the transaction and return the EVM's execution result.
-	execResult, err := k.miner.ProcessTransaction(ctx, tx)
+	receipt, err := k.miner.ProcessTransaction(ctx, tx)
 	if err != nil {
 		return nil, err
 	}
 
-	// We don't want the cosmos transaction to be marked as failed if the EVM reverts. But
-	// its not the worst idea to log the error.
-	if execResult.Err != nil {
-		k.Logger(sdk.UnwrapSDKContext(ctx)).Error(
-			"evm execution",
-			"tx_hash", tx.Hash(),
-			"error", execResult.Err,
-			"gas_consumed", sCtx.GasMeter().GasConsumed(),
-		)
```

Comment:
this case as well

Reply:
nah let's simplify we shouldn't care, this approach reduces the geth fork by about 100 lines